en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/defining-interceptors-in-an-open-api-definition.md

```diff
@@ -0,0 +1,245 @@
+overridden by the resource level interceptor for that resource.
+
+For an API, you can define
+
+- one of request or response interceptor
+- or both of request and response interceptor
+
+base on your requirement for the mediation.
+
+
+## Request flow Interceptor
+
+You can define the request flow interceptor as follows in the Open API Definition with the extension `x-wso2-request-interceptor`.
+[Description of Keys of Interceptor Open API Extension](#description-of-keys-of-interceptor-open-api-extension) explains
+detailed description of the keys.
+
+``` yaml tab="Format"
+x-wso2-request-interceptor:
+  scheme: <one of [http|https]>
+  host: <host>
+  port: <port>
+  includes: # any of following
+    - request_headers
+    - request_body
+    - request_trailers
+    - invocation_context
+  requestTimeout: OPTIONAL <request timeout in seconds>
+  clusterTimeout: OPTIONAL <override interceptor cluster timeout in seconds>
+```
+
+``` yaml tab="Example API Level"
+.
+.
+info:
+  version: 1.0.5
+  title: PizzaShackAPI
+x-wso2-basePath: /v2
+x-wso2-production-endpoints:
+    urls:
+    - https://localhost:2380/v2
+x-wso2-request-interceptor:
+  scheme: http
+  host: host.request.interceptor
+  port: 9081
+  includes:
+    - request_headers
+    - invocation_context
+.
+.
+```
+
+``` yaml tab="Example Resource Level"
+.
+.
+info:
+  version: 1.0.5
+  title: PizzaShackAPI
+x-wso2-basePath: /v2
+x-wso2-production-endpoints:
+    urls:
+    - https://localhost:2380/v2
+paths:
+    /pet/findByStatus:
+        x-wso2-production-endpoints:
+            urls:
+            -  https://localhost:2380/v1
+        x-wso2-request-interceptor:
+            scheme: http
+            host: host.request.interceptor
+            port: 9081
+            includes:
+            - request_body
+        get:
+            tags:
+            - pets
+            summary: Finds Pets by status
+            description: Multiple status values can be provided with comma separated strings
+            operationId: findPetsByStatus
+            parameters:
+            - name: status
+              in: query
+              description: Status values that need to be considered for filter
+        .
+        .
+    /pet/{petId}:
+        get:
+.
+.
+```
+
+<!-- The content of the below warning is same as the info notice in the file
+deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/request-flow-interceptor.md -->
+!!! info
+    If you update the request body before reaching the backend, ensure to add `request_body` to the `includes` section
+    of the Open API Specification. You may have a scenario like, whatever the request body from the client, you define
+    your own request body (say, based on a header value) to send to the backend. Even though you do not read the
+    request body, you should do the above inclusion.
```

Comment:
If you update the request body before reaching the backend, ensure to add `request_body` to the `includes` section of the Open API Specification. Your scenario might be that regardless of the request body from the client, you define your own request body (say, based on a header value) to send to the backend. Even though you do not read the request body, you should include the above condition.